Source/WTF/wtf/GenerateProfiles.h

```diff
@@ -59,18 +59,19 @@ ALWAYS_INLINE void registerProfileGenerationCallback(const char* name)
             "`.");
         WTFLogAlways("<WEBKIT_LLVM_PROFILE> To sanity-check the output: `for f in ./profiles/*; do echo $f; xcrun -sdk macosx.internal llvm-profdata show $f; done;`.");
 
-        {
-            // Maybe we could use %t instead here, but this folder is permitted through the sandbox because of ANGLE.
-            FileSystem::PlatformFileHandle fileHandle;
-            auto filePath = FileSystem::openTemporaryFile(makeString(name, "-", getpid()), fileHandle, ".profraw"_s);
-            profileFileBase.get() = String::fromUTF8(filePath.utf8().data());
-            FileSystem::closeFile(fileHandle);
-        }
-
         WTFLogAlways("<WEBKIT_LLVM_PROFILE><%s><%d>: We will dump the resulting profile to %s.", name, pid, profileFileBase->utf8().data());
 
         int token;
         notify_register_dispatch("com.apple.WebKit.profiledata", &token, dispatch_get_main_queue(), ^(int) {
+            
+            {
+                // Maybe we could use %t instead here, but this folder is permitted through the sandbox because of ANGLE.
```

Comment:
I know that this is just moved code, but there are couple things that I don't understand.
- Why is this is a block? Doesn't seem like there is any object that we must destroy at the end of the block.
- What is this comment saying about ANGLE, or how that's related to %t.
- What are we achieving by creating a temporary file, and immediately closing it. Is it just that FileSystem doesn't have the necessary primitives, and we work around that in this way?

Reply:
Discussed with Alexey in person, let's check in this change as is for now. We will reiterate this with Justin.

Reply:
This is in a block so we don't use the file handle after it is closed. There is already space in the sandbox at that location because ANGLE writes some shader stuff there.

Creating the file was not a super principled thing. It was handy while debugging to have the process crash early if it couldn't write the files, instead of waiting until we try to collect the profiles. This way, I was certain that profiles weren't getting silently missed. It's not necessarily the best or right way of doing things.

As for %t, if we had an extra sandbox exception (or change the TEMPDIR environment variable), we could just use llvm's substitution. The comment doesn't make this clear. See https://clang.llvm.org/docs/SourceBasedCodeCoverage.html